cluster/definition.yaml

```diff
@@ -105,6 +105,10 @@ spec:
               id:
                 type: string
                 description: ID of this Cluster that other objects will use to refer to it.
+                minLength: 1
+                # Maximum key length in GCP is 40 characters. We are setting a 34 character
+                # limit to accomodate the generated hash suffix ('-' + 5 chars).
+                maxLength: 34
```

Comment:
Interesting to think about validation at the top-level that just adheres to one "implementation" (i.e. Composition that references the XRD). It would be cool for us to add the ability for Crossplane webhooks to do admission validation recursively on embedded schemas (i.e. if the GCP Composition was going to be selected in this case, we could reject it based on how the field would propagate to the GKE cluster XR, but not reject if a different Composition was selected).

cc @negz @muvaf @tnthornton

Reply:
Something like `up plan` that can run dry-run the whole composition flow could also catch it earlier.